src/test/java/seedu/address/logic/commands/UnmatchCommandTest.java

```diff
@@ -0,0 +1,182 @@
+ */
+public class UnmatchCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+    }
+
+    @Test
+    public void constructor_nullIndex_throwsNullPointerException() {
+        Index validIndex = Index.fromOneBased(1);
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(null, null));
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(null, validIndex));
+        assertThrows(NullPointerException.class, () -> new UnmatchCommand(validIndex, null));
+    }
+
+    @Test
+    public void execute_validContactAndJob_unmatchSuccessful() throws Exception {
+        Index contactIndex = INDEX_FIRST_PERSON;
+        Index jobIndex = INDEX_FIRST_JOB;
+
+        // First, set up the model with an existing match
+        setModelAfterMatching(model, contactIndex, jobIndex);
+
+        UnmatchCommand unmatchCommand = new UnmatchCommand(contactIndex, jobIndex);
+
+        // Set up the expected model state after unmatching
+        Person contactInExpectedModel = expectedModel.getFilteredPersonList().get(contactIndex.getZeroBased());
+        Person unmatchedContact = new PersonBuilder(contactInExpectedModel).withoutMatch().build();
+        expectedModel.setPerson(contactInExpectedModel, unmatchedContact);
+
+        // Format the expected success message
+        String expectedMessage = String.format(UnmatchCommand.MESSAGE_UNMATCH_SUCCESS,
+                Messages.format(unmatchedContact), Messages.format(expectedModel
+                        .getFilteredJobList().get(jobIndex.getZeroBased())));
+
+        // Verify the command execution and resulting state
+        assertCommandSuccess(unmatchCommand, model, expectedMessage, expectedModel);
+    }
+
+
+    @Test
+    public void execute_contactIndexOutOfRange_failure() {
+        UnmatchCommand unmatchCommand = new UnmatchCommand(INDEX_OUT_OF_BOUNDS, INDEX_FIRST_JOB);
+        assertCommandFailure(unmatchCommand, model, Messages.MESSAGE_INVALID_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_jobIndexOutOfRange_failure() {
+        UnmatchCommand unmatchCommand = new UnmatchCommand(INDEX_FIRST_PERSON, INDEX_OUT_OF_BOUNDS);
+        assertCommandFailure(unmatchCommand, model, Messages.MESSAGE_INVALID_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_emptyModel_failure() {
+        UnmatchCommand unmatchCommand = new UnmatchCommand(INDEX_FIRST_PERSON, INDEX_FIRST_JOB);
+        Model emptyModel = new ModelManager(new AddressBook(), new UserPrefs());
+        assertCommandFailure(unmatchCommand, emptyModel, Messages.MESSAGE_INVALID_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_contactNotMatched_failure() {
```

Comment:
The test only works if  your assumption of first contact and first in not already matched is true. 

Perhaps an assertion can be used, you could also consider a helper function that unmatch a pair (if matched), or add an unmatched pair at the start of the test.

Reply:
I think it should be fine as a new model with a sample address book is used for each test case, meaning that the first person is always unmatched.

Reply:
Although the model will be "refreshed" for each test, there is no guarantee that the custom getAddressBook would be the same in the future. 

That custom function depends on `getTypicalPerson()`, what if the in the future, `getTypicalPerson()` return a list where first person is already matched? Then then assumption of this test will be wrong.